Makefile

```diff
@@ -1,133 +1,24 @@
-
-compile-test: compile target/.java-test
-
-TEST_CLASSES=$(shell build/get-test-classes)
-
-define get_test_parameters
-	$(eval filter=$(shell echo $(1) | perl -ne '/#(.*)/; print "--filter=com.zendesk.maxwell.JUnitNameFilterFactory=$$1" if $$1')) \
-	$(eval class_pattern=$(shell echo $(1) | perl -ne '/test\.(\w+)/; print $$1 if $$1')) \
-	$(eval classes=$(if $(class_pattern),$(filter %$(class_pattern),$(TEST_CLASSES)),$(TEST_CLASSES))) \
-	$(filter) $(classes)
-endef
-
-JAVA_TEST_PARAMS=-Djava.util.logging.config.file=./src/test/resources/logging.properties
+all: compile
 
-test%: compile-test
-	java $(JAVA_TEST_PARAMS) -classpath $(JAVA_TEST_DEPENDS):target/test-classes:target/classes org.junit.runner.JUnitCore $(call get_test_parameters,$@)
+test:
+	mvn test -P ${KAFKA_PROFILE}
 
-test: compile-test
-	java $(JAVA_TEST_PARAMS) -classpath $(JAVA_TEST_DEPENDS):target/test-classes:target/classes org.junit.runner.JUnitCore $(TEST_CLASSES)
-test-only: compile-test
-	java $(JAVA_TEST_PARAMS) -classpath $(JAVA_TEST_DEPENDS):target/test-classes:target/classes org.junit.runner.JUnitCore ${ARGS}
+compile:
+	mvn compile -P ${KAFKA_PROFILE}
 
 clean:
-	rm -f  target/.java target/.java-test
-	rm -rf target/classes
-	rm -rf target/generated-sources
-	rm -f  target/*.jar
+	mvn clean
 
 depclean: clean
 	rm -f $(CLASSPATH)
-	rm -Rf target/dependency
-
-PKGNAME=maxwell-${MAXWELL_VERSION}
-MAVEN_DIR=target/classes/META-INF/maven/com.zendesk/maxwell
-MAXWELL_JARFILE=target/$(PKGNAME).jar
-
-package-jar: all
-	rm -f ${MAXWELL_JARFILE}
-	mkdir -p ${MAVEN_DIR}
-	sed -e "s/VERSION/${MAXWELL_VERSION}/" build/pom.properties > ${MAVEN_DIR}/pom.properties
-	cp pom.xml ${MAVEN_DIR}
-	echo "Implementation-Version: ${MAXWELL_VERSION}" >target/jar-manifest
-	jar cvmf target/jar-manifest ${MAXWELL_JARFILE} -C target/classes .
-
-TARDIR=target/$(PKGNAME)
-TARFILE=target/$(PKGNAME).tar.gz
-
-package-tar:
-	rm -Rf target/dependency-build
-	build/maven_fetcher -p -o target/dependency-build >/dev/null
- 	# Include kafka 0.8 jar
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_08) --skip-dependencies -o target/dependency-build >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_010) --skip-dependencies -o target/dependency-build >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_01010) --skip-dependencies -o target/dependency-build >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_01021) --skip-dependencies -o target/dependency-build >/dev/null
-	rm -Rf $(TARDIR) $(TARFILE)
-	mkdir $(TARDIR)
-	cp $(DISTFILES) $(TARDIR)
-	cp -a bin $(TARDIR)
-	mkdir $(TARDIR)/lib
-	cp -a $(MAXWELL_JARFILE) target/dependency-build/* $(TARDIR)/lib
-	tar czvf $(TARFILE) -C target $(PKGNAME)
-
-package: depclean package-jar package-tar
 
+package: depclean
+	@# TODO: this is inefficient, we really just want to copy the jars...
```

Comment:
What if we depend on these versions in `dependencies`?

Reply:
I tried that - maven complains if you depend on multiple versions of a dependency, and picks the latest. Which is usually what you want, but we're doing something weird here on purpose.